### PR TITLE
Rename to cee-rhel7 (ATDV-391), remove last of REHL6 (ATDV-404), add sems-rhel7 clang build (ATDV-291), cee-rhel7 support for SPARC TPL installer (ATDV-408) and more

### PR DESCRIPTION
**Pull request by bartlettroscoe**
This PR addresses several issues mostly driven by getting the ATDM Trilinos system used by the SPARC TPL  installer ([ATDV-408](https://sems-atlassian-srn.sandia.gov/browse/ATDV-408)).  See the [commits messages](https://github.com/trilinos/Trilinos/pull/8525/commits) for the context and links to driving stories.  Here are all of the issues addressed by this PR: 

* [ATDV-291](https://sems-atlassian-srn.sandia.gov/browse/ATDV-291): ATDM Trilinos sems-rhel7 clang-7 support for EMPIRE
* [ATDV-391](https://sems-atlassian-srn.sandia.gov/browse/ATDV-391): Change name of ATDM Trilinos 'cee-rhel6' env to 'cee-rhel7'
* [ATDV-404](https://sems-atlassian-srn.sandia.gov/browse/ATDV-404): Remove the 'sems-rhel6' automated builds and env
* [ATDV-408](https://sems-atlassian-srn.sandia.gov/browse/ATDV-408): Transition SPARC to use ATDM Trilinos configuration (only 'cee-rhel7' is addressed in this PR branch.)

## How was this tested?

See comments in [ATDV-291](https://sems-atlassian-srn.sandia.gov/browse/ATDV-291), [ATDV-391](https://sems-atlassian-srn.sandia.gov/browse/ATDV-391), [ATDV-408](https://sems-atlassian-srn.sandia.gov/browse/ATDV-408)

### CODE REVIEW

**Review by william76 — APPROVED** [review]

**Review by william76 — APPROVED** [review]

**Comment by bartlettroscoe** [comment]
Thanks @william76!

**Comment by bartlettroscoe** [comment]
Hum, the autotester said it was stale yet I was able to click "merge".  So this is merged.

File: cmake/std/atdm/utils/get_known_system_info.sh [hunk 1]
```diff
@@ -72,8 +72,7 @@ ATDM_KNOWN_SYSTEM_NAMES_LIST=(
   cts1
   tlcc2
   sems-rhel7
-  sems-rhel6
-  cee-rhel6  # Used for CEE RHEL7 machines as well!
+  cee-rhel7  # Used for CEE RHEL7 machines as well!
```
Comment (jmgate) [inline cmake/std/atdm/utils/get_known_system_info.sh:75]: Probably want to remove the comment.

**Review by jmgate — COMMENTED** [review]
Looks good, excepting one comment that should be removed.  Thanks @bartlettroscoe.

File: cmake/std/atdm/utils/get_known_system_info.sh (hunk 1 above)
Comment (bartlettroscoe) [reply cmake/std/atdm/utils/get_known_system_info.sh:75]: Thanks for catching that @jmgate.  I will fix in a new PR.

**Comment by bartlettroscoe** [comment]
Related to:
* SEPW-214

File: cmake/std/atdm/README.md [hunk 2]
```diff
@@ -1559,10 +1566,10 @@ configuration script.
 
 A simple example can be seen in:
 
-* [cmake/std/atdm/examples/my-sems-rhel6-config/environment.sh](examples/my-sems-rhel6-config/environment.sh)
+* [cmake/std/atdm/examples/my-sems-rhel7-config/environment.sh](examples/my-sems-rhely-config/environment.sh)
```
Comment (e10harvey) [inline cmake/std/atdm/README.md:1569]: typo: rhely->rhel7

File: cmake/std/atdm/test/unit_tests/set_build_options_unit_tests.sh [hunk 3]
```diff
@@ -370,6 +372,22 @@ testRDC() {
...
+  ${_ASSERT_EQUALS_} OFF ${ATDM_CONFIG_ADDRESS_SANITIZER}
+
+  ATDM_CONFIG_BUILD_NAME=default-asan-after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ON ${ATDM_CONFIG_ADDRESS_SANITIZER}
+
+  ATDM_CONFIG_BUILD_NAME=default_asan-after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ON ${ATDM_CONFIG_ADDRESS_SANITIZER}
+
```
Comment (e10harvey) [inline cmake/std/atdm/test/unit_tests/set_build_options_unit_tests.sh:388]: Missing tests for `no-asan`?

**Review by e10harvey — COMMENTED** [review]
Thanks, @bartlettroscoe !

**Pull request merged**